pkg/policy/l4.go

```diff
@@ -783,32 +789,43 @@ const (
 	OriginatingTLS TLSDirection = "originating"
 )
 
+// getCerts reads certificates out of the PolicyContext, reading from k8s or local files depending on config
+// and puts the values into the relevant keys in the TLSContext. Note that if the returned TLSContext.FromFile is
+// `false`, then this will be read from Kubernetes.
 func (l4 *L4Filter) getCerts(policyCtx PolicyContext, tls *api.TLSContext, direction TLSDirection) (*TLSContext, error) {
 	if tls == nil {
 		return nil, nil
 	}
-	ca, public, private, err := policyCtx.GetTLSContext(tls)
+	ca, public, private, inlineSecrets, err := policyCtx.GetTLSContext(tls)
 	if err != nil {
 		log.WithError(err).Warningf("policy: Error getting %s TLS Context.", direction)
 		return nil, err
 	}
-	switch direction {
-	case TerminatingTLS:
-		if public == "" || private == "" {
-			return nil, fmt.Errorf("Terminating TLS context is missing certs.")
-		}
-	case OriginatingTLS:
-		if ca == "" {
-			return nil, fmt.Errorf("Originating TLS context is missing CA certs.")
+
+	// If the secret is not being included into NPDS inline, we're going to pass an SDS reference instead.
+	if inlineSecrets {
+		switch direction {
+		case TerminatingTLS:
+			if public == "" || private == "" {
+				return nil, fmt.Errorf("Terminating TLS context is missing certs.")
+			}
+		case OriginatingTLS:
+			if ca == "" {
+				return nil, fmt.Errorf("Originating TLS context is missing CA certs.")
+			}
+		default:
+			return nil, fmt.Errorf("invalid TLS direction: %s", direction)
 		}
-	default:
-		return nil, fmt.Errorf("invalid TLS direction: %s", direction)
+	} else {
+		log.Debug("Secret being read from Kubernetes", "secret", types.NamespacedName(*tls.Secret))
```

Comment:
is this else case and the logging necessary? It looks like it's a duplication as the same message is already logged in `(m *manager) GetTLSContext` (https://github.com/cilium/cilium/blob/47556bd245d901b0fd89da7e830f2acbe68c0709/pkg/crypto/certificatemanager/certificate_manager.go#L170)